dev/com.ibm.ws.jaxrs.2.x.monitor/src/com/ibm/ws/jaxrs/monitor/JaxRsMonitorFilter.java

```diff
@@ -0,0 +1,238 @@
+import javax.ws.rs.ext.Provider;
+
+import com.ibm.websphere.csi.J2EEName;
+import com.ibm.websphere.monitor.annotation.Monitor;
+import com.ibm.websphere.monitor.annotation.PublishedMetric;
+import com.ibm.websphere.monitor.meters.MeterCollection;
+import com.ibm.ws.runtime.metadata.ComponentMetaData;
+import com.ibm.ws.runtime.metadata.ModuleMetaData;
+import com.ibm.ws.threadContext.ComponentMetaDataAccessorImpl;
+
+
+/**
+ * Monitor Class for RESTful Resource Methods.
+ */ 
+@Monitor(group = "RESTful")
+@Provider
+public class JaxRsMonitorFilter implements ContainerRequestFilter, ContainerResponseFilter {
+
+    @Context
+    ResourceInfo resourceInfo;
+    
+    // jaxRSCountByName is a MeterCollection that will hold the RESTfulStats MXBean for each RESTful
+    // resoure method
+    @PublishedMetric
+    public MeterCollection<RESTful_Stats> jaxRsCountByName = new MeterCollection<RESTful_Stats>("RESTful",this);
+    
+    // appMetricInfos is a hashmap used to store information for runtime and cleanup at 
+    // application stop time.
+    protected ConcurrentHashMap<String,RestfulMetricInfo> appMetricInfos = new ConcurrentHashMap<String,RestfulMetricInfo>();
+    
+    private final ThreadLocal<Long> startTimes = new ThreadLocal<Long>();    
+
+    /**
+     * Method : filter(ContainerRequestContext)
+     * 
+     * @param reqCtx = Container request context
+     * 
+     *            This method will be called whenever a request arrives for a
+     *            RESTful resource method.
+     *            All it does is save the current time so that the response
+     *            time can be calculated later.
+     * 
+     */
+    @Override
+    public void filter(ContainerRequestContext reqCtx) throws IOException {
+    	
+        startTimes.set(System.nanoTime());
+  	
+    }
+    /**
+     * Method : filter(ContainerRequestContext, ContainerResponseContext)
+     * 
+     * @param reqCtx = Container request context
+     * @param respCtx = Container response context
+     * 
+     *            This method will be called whenever a response arrives from a
+     *            RESTful resource method.
+     *            It will calculate the cumulative response time for the resource
+     *            method and store the result in the RESTful_Stats MXBean.
+     * 
+     */
+
+    @Override
+    public void filter(ContainerRequestContext reqCtx, ContainerResponseContext respCtx) throws IOException {
+        Class<?> resourceClass = resourceInfo.getResourceClass();
+        if (resourceClass != null) {
+            Method resourceMethod = resourceInfo.getResourceMethod();
+            String resourceMethodString = resourceMethod.toGenericString();
+            
+            //Spit to obtain just the fully qualified method name (without app and mod information).
+            String[] methodParts = resourceMethodString.split(" ");
+            String simpleMethodName = methodParts[methodParts.length -1];
```

Comment:
I had looked at that, but the problem is that resourceMethod.getName() only returns the non-fully qualified method name (ie. listBooks) whereas I wanted the fully qualified name including parameters (ie. io.lilfreelib.BookResource.listBooks()).   Perhaps there is an easier way to get this, but I didn't find it.

Reply:
I think you'll still need to get the fully-qualified class name from the class and then add `"." + resourceMethod.getName()` to it.  Otherwise, if the resource method throws an exception your method name will be off - see from the javadoc:
> If the method is declared to throw exceptions, the parameter list is followed by a space, followed by the word throws followed by a comma-separated list of the generic thrown exception types.

thus:
```
@GET
public MyObject getMyObject(@PathParam("{objectId}") String objectId) throws InvalidObjectIDException { ...  
```

would end up returning "InvalidObjectIDException" for the `simpleMethodName`.